base/applications/sndrec32/sndrec32.cpp

```diff
@@ -889,14 +935,71 @@ WndProc(HWND hWnd,
     return 0;
 }
 
-void l_play_finished(void)
+void DrawButton(int ButtonID, LPARAM lParam)
+{
+    LPDRAWITEMSTRUCT lpdis = (DRAWITEMSTRUCT*)lParam;
+    SIZE size;
+    LRESULT ButtonStatus;
+    int ButtonStyle = DFCS_BUTTONPUSH;
+    int lpdx[] = {8,8}, captionspacer=0;
+
+    if (ButtonID<BUTPLAY_ID)
+    {
+        captionspacer=8;
+    }
+
+
+    if (ButtonID >= BUTREC_ID)
+    {
+        SetTextColor(lpdis->hDC, RGB(255, 0, 0));
+    }
+    else
+    {
+        SetTextColor(lpdis->hDC, RGB(0, 0, 0));
+    }
+
+    ButtonStatus = SendMessage(buttons[ButtonID], BM_GETSTATE, NULL, NULL);
```

Comment:
What's going to happen in this `DrawButton` function if one calls it with a button ID that is >= BUTREC_ID ?

Reply:
it sets the text color to read, for the record button. I used >= because BUTREC_ID is the bigges ID provided, but I think a == would be more correct.

Reply:
Yes, but if one calls the `DrawButton` function with a button ID that is > BUTREC_ID (greater strictly), it will throw an access violation because then you're going to access the buttons array beyond its end.

Reply:
OK. I guarded the call:
```
if ((ButtonID >= BUTSTART_ID) && (ButtonID <= BUTREC_ID))
{
    ButtonStatus = SendMessage(buttons[ButtonID], BM_GETSTATE, NULL, NULL);
}
```